src/components/m-button/mixins/_button.scss

```diff
@@ -135,16 +129,35 @@
   padding-bottom: $button-padding-top-bottom-lg;
 }
 
-@mixin button__icon($x-padding: $button-x-padding-default) {
-  $local-padding: 8px;
+@mixin button--generic-icon--sm($x-padding: $button-x-padding-default) {
   $x-padding: min-max($button-x-padding-min, $button-x-padding-max, $x-padding);
+  padding-right: $x-padding + $button-sm-height;
+}
+
+@mixin button--generic-icon($x-padding: $button-x-padding-default) {
+  $x-padding: min-max($button-x-padding-min, $button-x-padding-max, $x-padding);
+  padding-right: $x-padding + $button-md-height;
+}
+
+@mixin button--generic-icon--lg($x-padding: $button-x-padding-default) {
+  $x-padding: min-max($button-x-padding-min, $button-x-padding-max, $x-padding);
+  padding-right: $x-padding + $button-lg-height;
+}
+
+@mixin button__icon() {
+  $local-padding: 8px;
 
   @include size($button-md-height);
 
-  margin-left: $x-padding;
+  position: absolute;
+
+  top: 0;
+  right: 0;
 
-  margin-top: - $x-padding / 2;
-  margin-bottom: - $x-padding / 2;
+  margin-top: - $button-border-width;
+  margin-bottom: 0;
+
+  height: calc(100% + #{$button-border-width});
```

Comment:
I thought if you omit `height` and use `top: -$button-border-width;` and `bottom: -$bottom-border-width` it should stretch by itself